[video][web] Fix `AudioContext` based audio management issues

behenate opened the pull request:
# Why

There were errors thrown on fast-refresh and refresh when using `expo-video` on web. This was due to incorrect audio nodes management. `AudioContext.createMediaElementSource` was called multiple times per `<video>` element, which is not allowed. Also during a refresh due to changes in `VideoView.web.tsx` the refresh would fail, because new AudioContext was created, but the video was still bound to the previous one.

# How

Moved the AudioContext creation into the VideoView component. This means that each VideoView has it's own AudioContext that will stay the same throughout the lifecycle of the component.

In external apps `expo-video` on web was throwing an error due to checking `window && ...`. Changed this check to `typeof window !== 'undefinded`

# Test Plan

Texted in NCL on web and in a separate app.

behenate committed:
[video][web] Fix audio management

tsapeta reviewed (approved):
Looks good to me, but let's wait for @aleqsio review

behenate committed:
[video] Separate `VideoView.tsx` into `VideoView.tsx` and `VideoPlayer.tsx` (#28225)

# Why

Made the project structure a bit more clear.

# Test Plan

Tested in NCL on web and in BareExpo on iOS

aleqsio reviewed (approved):
Tried to find something that wouldn't work here, but it seems alright 😛  Thanks @behenate 💛

behenate committed:
Update the docs files, review suggestions
